bundles/core/org.eclipse.smarthome.core.semantics/src/main/java/org/eclipse/smarthome/core/semantics/internal/SemanticsMetadataProvider.java

```diff
@@ -61,21 +61,9 @@
     public static final String NAMESPACE = "semantics";
 
     // holds the static definition of the relations between entities
-    private static Map<List<Class<? extends Tag>>, String> RELATIONS_PARENT = new HashMap<>();
-    private static Map<List<Class<? extends Tag>>, String> RELATIONS_MEMBER = new HashMap<>();
-    private static Map<List<Class<? extends Tag>>, String> RELATIONS_PROPERTY = new HashMap<>();
-
-    static {
-        RELATIONS_PARENT.put(Arrays.asList(Equipment.class, Location.class), "hasLocation");
-        RELATIONS_PARENT.put(Arrays.asList(Point.class, Location.class), "hasLocation");
-        RELATIONS_PARENT.put(Arrays.asList(Location.class, Location.class), "isPartOf");
-        RELATIONS_PARENT.put(Arrays.asList(Equipment.class, Equipment.class), "isPartOf");
-        RELATIONS_PARENT.put(Arrays.asList(Point.class, Equipment.class), "isPointOf");
-
-        RELATIONS_MEMBER.put(Arrays.asList(Equipment.class, Point.class), "hasPoint");
-
-        RELATIONS_PROPERTY.put(Arrays.asList(Point.class), "relatesTo");
-    }
+    private final Map<List<Class<? extends Tag>>, String> RELATIONS_PARENT = new HashMap<>();
```

Comment:
Where's the benefit to have this really very static information in fields? If it is about making sure to have clean maps, shouldn't you maybe also clear the maps upon deactivation?
When changing it to fields, you should not use capitalised names anymore.

Reply:
I have to admit, I did this in the light of our recommendation to not have loggers defined static. I just realised that this is a recommendation from SLF4J where they say

> instance variables are IOC-friendly whereas static variables are not

w/o any further explanation.

I guess static would be equally okay now.

> shouldn't you maybe also clear the maps upon deactivation

I don't think so as the service instance will be disposed and garbage collected anyhow (thats at least my understanding). Initialising in `activate` was mainly done due to the lack of a constructor.

> When changing it to fields, you should not use capitalised names anymore.

true, will change that.